homeassistant/components/stt/__init__.py

```diff
@@ -36,33 +49,171 @@
+            raise RuntimeError("Entity is not added to hass yet.")
+
+        return self.platform.platform_name
+
+    @property
+    @final
+    def state(self) -> str | None:
+        """Return the state of the provider entity."""
+        if self.__last_processed is None:
+            return None
+        return self.__last_processed
+
+    @property
+    @abstractmethod
+    def supported_languages(self) -> list[str]:
+        """Return a list of supported languages."""
+
+    @property
+    @abstractmethod
+    def supported_formats(self) -> list[AudioFormats]:
+        """Return a list of supported formats."""
+
+    @property
+    @abstractmethod
+    def supported_codecs(self) -> list[AudioCodecs]:
+        """Return a list of supported codecs."""
+
+    @property
+    @abstractmethod
+    def supported_bit_rates(self) -> list[AudioBitRates]:
+        """Return a list of supported bit rates."""
+
+    @property
+    @abstractmethod
+    def supported_sample_rates(self) -> list[AudioSampleRates]:
+        """Return a list of supported sample rates."""
+
+    @property
+    @abstractmethod
+    def supported_channels(self) -> list[AudioChannels]:
+        """Return a list of supported channels."""
+
+    async def async_internal_added_to_hass(self) -> None:
+        """Call when the provider entity is added to hass."""
+        await super().async_internal_added_to_hass()
+        state = await self.async_get_last_state()
+        if (
+            state is not None
+            and state.state is not None
+            and state.state not in (STATE_UNAVAILABLE, STATE_UNKNOWN)
+        ):
+            self.__last_processed = state.state
+
+    @final
+    async def internal_async_process_audio_stream(
+        self, metadata: SpeechMetadata, stream: AsyncIterable[bytes]
+    ) -> SpeechResult:
+        """Process an audio stream to STT service.
+
+        Only streaming content is allowed!
+        """
+        self.__last_processed = dt_util.utcnow().isoformat()
+        self.async_write_ha_state()
+        return await self.async_process_audio_stream(metadata=metadata, stream=stream)
+
+    @abstractmethod
+    async def async_process_audio_stream(
+        self, metadata: SpeechMetadata, stream: AsyncIterable[bytes]
+    ) -> SpeechResult:
+        """Process an audio stream to STT service.
+
+        Only streaming content is allowed!
+        """
+
+    @callback
+    def check_metadata(self, metadata: SpeechMetadata) -> bool:
+        """Check if given metadata supported by this provider."""
+        if (
+            metadata.language not in self.supported_languages
+            or metadata.format not in self.supported_formats
+            or metadata.codec not in self.supported_codecs
+            or metadata.bit_rate not in self.supported_bit_rates
+            or metadata.sample_rate not in self.supported_sample_rates
+            or metadata.channel not in self.supported_channels
+        ):
+            return False
```

Comment:
This copied from the legacy implementation, but how does it work really?

Not all permutations of the settings are possibly supported, in particular the supported bit rates are not same between opus and raw pcm.

Reply:
Ok, I don't know more about this unfortunately.